go/vt/vtgate/engine/delete.go

```diff
@@ -145,6 +145,24 @@ func (del *Delete) RouteType() string {
 	return delName[del.Opcode]
 }
 
+// KeyspaceTableNames specifies the table that this primitive routes to
+func (del *Delete) KeyspaceTableNames() []*KeyspaceTableName {
+	if del.Table != nil {
```

Comment:
We don't do much analysis for unsharded tables. So, this field doesn't get set. We should ideally populate this for all cases, eventually.